### PR TITLE
Fixes #8563: Make the search bar in menu able to also search rule, group, directive, etc

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/8563

This is a work in progress with the current prototype, to allow some pre-validation (try the feature by yourself!) and looking to the code for architecture remarks.

### CODE REVIEW

**Comment by fanf** [comment]
PR rebased

File: rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+
+  /**
+   * Lookup directives
+   */
+  def search(query: Query)(implicit repo: RoDirectiveRepository): Box[Set[QuickSearchResult]] = {
+
+    // only search if query is on Directives and attributes contains
+    // DirectiveId, DirectiveVarName, DirectiveVarValue, TechniqueName, TechniqueVersion
+
+    val attributes: Set[QSAttribute] = query.attributes.intersect(QSObject.Directive.attributes)
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala]: QSObject.Directive can be replaced by QSDirective

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

File: rudder-web/src/main/scala/com/normation/rudder/appconfig/ConfigService.scala [hunk 2]
```diff
@@ -62,6 +62,33 @@ import com.normation.rudder.domain.eventlog.ModifyAgentRunStartMinuteEventType
...
+import ca.mrvisser.sealerate
+import com.normation.rudder.web.components.popup.ModificationValidationPopup.Disable
+
+
+/**
+ * A little domain language for feature switches
+ * (just enabled/disabled with the parsing)
+ */
+sealed trait FeatureSwitch { def name: String }
+object FeatureSwitch {
```
Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/appconfig/ConfigService.scala]: Use the implementation from PR  https://github.com/Normation/rudder/pull/1141

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

**Comment by VinceMacBuche** [comment]
PR rebased

**Comment by VinceMacBuche** [comment]
Replaced by #1188

**Pull request closed without merging**